### PR TITLE
server side implementation for store report service for report manager

### PR DESCRIPTION
**Pull request by Dmaddu**
Signed-off-by: Durga Sarat Chandra Maddu <durgasaratchandra.maddu@progress.com>

### :athletic_shoe: How to Build and Test the Change

from hab studio
rebuild components/report-manager-minio-gateway/
rebuild components/report-manager-service/
rebuild components/automate-cli/
rebuild components/automate-deployment/
start_all_services

`chef-automate status` should show process state as `running` and Health check should be `OK`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/report-manager-service/server/server.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	complianceReport := compliance.Report{}
+	err := json.Unmarshal(reportData.Bytes(), &complianceReport)
+	if err != nil {
+		return fmt.Errorf("error in converting report bytes to compliance report struct: %w", err)
+	}
+
+	objName := fmt.Sprintf("%s.json", complianceReport.GetReportUuid())
+
+	//TODO:: Add an expiry based on user configuration
+	info, err := s.MinioClient.PutObject(s.ctx, "default", objName, &reportData, -1, minio.PutObjectOptions{})
```
Comment (kalroy) [inline components/report-manager-service/server/server.go:75]: Will this handle the scenarios where the user has setup AWS, GCP or Azure for object store?

File: components/report-manager-service/reportmanager.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		return err
+	}
+	logrus.Infof("connection established to minio server, endPoint:%s", minioClient.EndpointURL())
+
+	return serveGrpc(ctx, conf, minioClient, connFactory)
+}
+
+func getMinioConnection(ctx context.Context, conf config.ReportManager) (*minio.Client, error) {
+
+	//TODO:: Get the below details from configuration
```
Comment (kalroy) [inline components/report-manager-service/reportmanager.go:35]: We need to have issues for every Todo

File: components/report-manager-service/reportmanager.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	minioClient, err := minio.New(endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV4(accessKeyID, secretAccessKey, ""),
+		Secure: useSSL,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error in creating a connection to minio server: %w", err)
+	}
+
+	//check if the default bucket is available or not
+	isExist, err := minioClient.BucketExists(ctx, "default")
```
Comment (kalroy) [inline components/report-manager-service/reportmanager.go:50]: Same concern--- will it work for Cloud providers too?

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/report-manager-service/reportmanager.go (hunk 3 above)
Comment (Dmaddu) [reply components/report-manager-service/reportmanager.go]: yes

**Pull request merged**